ospfd/ospfd.c

```diff
@@ -556,6 +556,8 @@ static void ospf_finish_final(struct ospf *ospf)
 		params = IF_DEF_PARAMS(ifp);
 		if (OSPF_IF_PARAM_CONFIGURED(params, if_area))
 			UNSET_IF_PARAM(params, if_area);
+
+		IF_DEF_PARAMS(ifp)->type = ospf_default_iftype(ifp);
```

Comment:
I'm not sure about this.

Other commands like `ip ospf hello-interval` are still preserved after a `no router ospf`. Why is `ip ospf network TYPE` special?

Reply:
Checking the other ospf configurations under interface, they are retained even after ospf instance is deleted. 
Should we remove all ospf configurations from interfaces ?

Reply:
I agree with @rwestphal here, the configurations under the interface should be preserved.   It is actually useful to keep all of those configs when you restart ospf you don't need to re issue all of the commands.

Reply:
This is especially useful if you  used the interface commands to enable ospf on the interface instead of using the "network"  commands under "router ospf"